datasets/_global/global_energy_own/crawler.py

```diff
@@ -0,0 +1,141 @@
+    "gogpt_retired_mw",
+    "gogpt_shelved_mw",
+    "gbpt_announced_mw",
+    "gbpt_construction_mw",
+    "gbpt_mothballed_mw",
+    "gbpt_operating_mw",
+    "gbpt_pre_construction_mw",
+    "gbpt_retired_mw",
+    "gbpt_shelved_mw",
+    "gbpt_cancelled_mw",
+    "gcmt_proposed_mtpa",
+    "gcmt_operating_mtpa",
+    "gcmt_shelved_mtpa",
+    "gcmt_mothballed_mtpa",
+    "gcmt_cancelled_mtpa",
+    "gspt_operating_ttpa",
+    "gspt_announced_ttpa",
+    "gspt_construction_ttpa",
+    "gspt_retired_ttpa",
+    "gspt_operating_pre_retirement_ttpa",
+    "gspt_mothballed_ttpa",
+    "gspt_cancelled_ttpa",
+    "total",
+]
+
+
+def crawl_company(context: Context, row: Dict[str, str]):
+    id = row.pop("entity_id")
+    name = row.pop("entity_name")
+    if name == "unknown":
+        return
+    original_name = row.pop("name_local", "")
+    reg_country = row.pop("registration_country", "")
+    legal_entity_id = row.pop("legal_entity_identifier", "")
+    entity_type = row.pop("entity_type", "")
+
+    if entity_type == "legal entity":
+        schema = "Company"
+    elif entity_type == "state body" or entity_type == "state":
+        schema = "PublicBody"
+    elif entity_type == "person":
+        schema = "Person"
+    else:
+        schema = "Company"  # 3 universities end up being companies
+    # full list of entity types:
+
+    # legal entity
+    # arrangement; very few, mostly look like companies
+    # state body
+    # state
+    # unknown entity
+    # person
+    entity = context.make(schema)
+    entity.id = context.make_slug(id)
+
+    entity.add("name", name)
+    entity.add("name", original_name)
+    entity.add("alias", row.pop("name_other", ""))
+    entity.add("weakAlias", row.pop("abbreviation", ""))
+    entity.add("idNumber", legal_entity_id)
+    entity.add("description", entity_type)
+    entity.add("country", reg_country)
+    entity.add("website", row.pop("home_page", ""))
+    if schema != "PublicBody" and schema != "Person":
+        entity.add(
+            "permId", row.pop("refinitiv_permid", "")
+        )  # some public bodies have permids ([warning  ] Unexpected data found   [global_energy_own] data={'refinitiv_permid': '5000049834'} dataset=global_energy_own)
+        entity.add("cikCode", row.pop("sec_central_index_key", ""))
+    address = h.make_address(
+        context,
+        country=reg_country,
+        state=row.pop("registration_subdivision", ""),
+        city=row.pop("headquarters_subdivision", ""),
+    )
+    h.copy_address(entity, address)
+
+    context.emit(entity)
+    context.audit_data(
+        row,
+        ignore=IGNORE,
+    )
+
+
+def crawl_rel(context: Context, row: Dict[str, str]):
+    index = row.pop("index")
+    subject_entity_id = row.pop("subject_entity_id")
+    interested_party_id = row.pop("interested_party_id")
+
+    ownership = context.make("Ownership")
+    ownership.id = context.make_id(subject_entity_id, interested_party_id, index)
```

Comment:
It looks like index is something that might change just based on the ordering in the table, not because of a meaning ful change in the data. I also don't think we're representing more than one relationship between a pair of entities. So It's best to remove index from the id.